.github/actions/build-dotnet/action.yml

```
@@ -137,11 +137,19 @@ runs:
         key: ${{ steps.detect-cache-configuration.outputs.key }}
         restore-keys: ${{ steps.detect-cache-configuration.outputs.restore-keys }}
 
+    - name: Create temporary global.json with desired .NET SDK version
+      shell: pwsh
+      working-directory: ${{ inputs.directory }}
+      run: |
+        "{ ""sdk"": { ""version"": ""$Env:DOTNET_SDK_TO_USE"" } }" > ./global.json
+
     - name: Build and Static Code Analysis
       id: build
       shell: pwsh
       working-directory: ${{ inputs.directory }}
       run: |
+        Write-Output ("Using .NET SDK from ./global.json: $(Get-Content .\global.json)")
```

```suggestion
        Write-Output "Using .NET SDK from ./global.json: $(Get-Content .\global.json)"
```

Also, a line break before the file content would be nice, because it may contain a formatted JSON, not just a single line.

---

This part is missing:

> Also, a line break before the file content would be nice, because it may contain a formatted JSON, not just a single line.

---

My bad.